pkgs/development/interpreters/wamr/default.nix

```diff
@@ -0,0 +1,44 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+}:
+
+stdenv.mkDerivation rec {
+  pname = "wamr";
+  version = "1.2.1";
+
+  src = fetchFromGitHub {
+    owner = "bytecodealliance";
+    repo = "wasm-micro-runtime";
+    rev = "WAMR-${version}";
+    hash = "sha256-RO5GBwiozdMaLt2m6rUlw+UohnYlba6llrVudAsIHW8=";
+  };
+  dontInstall = true;
+
+  nativeBuildInputs = [ cmake ];
+
+  configurePhase = ''
+    runHook preConfigure
+    mkdir -p build
+    cd build
+    cmake ../product-mini/platforms/linux
+    runHook postConfigure
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+    mkdir -p $out/bin
+    make
+    cp iwasm $out/bin
+    runHook postBuild
+  '';
+
+  meta = with lib; {
+    description = "WebAssembly Micro Runtime";
+    homepage = "https://github.com/bytecodealliance/wasm-micro-runtime";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ ereslibre ];
+    platforms = platforms.linux;
```

Comment:
I will improve this derivation to support more platforms, but I prefer to get it merged as it is first, and then iterate on it.

Reply:
Usually we use platforms to indicate the platforms a package _should_ work on, rather than the ones it currently does.  If it should work on a platform, but currently doesn't, that's what meta.broken is for.

Reply:
Thanks a lot for your reviews @alyssais; very appreciated. I updated the derivation.